### PR TITLE
DS-2466 results per pagein browse

### PR DESCRIPTION
**Pull request by royopa**
https://jira.duraspace.org/browse/DS-2466

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/browse/BrowserScope.java [hunk 1]
```diff
@@ -354,7 +360,7 @@ public int getResultsPerPage()
      */
     public void setResultsPerPage(int resultsPerPage)
     {
-        if (resultsPerPage > -1 || (browseIndex != null && browseIndex.isTagCloudEnabled()))
```
Comment (helix84) [inline dspace-api/src/main/java/org/dspace/browse/BrowserScope.java]: This change is essentially a revert of the previous commit. Is it intentional or were you accidentally working with older source?

**Comment by helix84** [comment]
Just to be clear - this change affects the default number of items per page in browse pages, both XMLUI and JSPUI, both Discovery and Lucene, correct?

File: dspace-api/src/main/java/org/dspace/browse/BrowserScope.java (hunk 1 above)
Comment (royopa) [inline dspace-api/src/main/java/org/dspace/browse/BrowserScope.java]: I worked with an older source, did the fix in commit
https://github.com/quatroelementos/DSpace/commit/3f74f3cc24c1a988381e8c0d25405c6fef45ca1f
thanks

On Tue, Feb 24, 2015 at 5:33 AM, helix84 <notifications@github.com> wrote:

> In dspace-api/src/main/java/org/dspace/browse/BrowserScope.java
> <https://github.com/DSpace/DSpace/pull/872#discussion_r25235500>:
>
> > @@ -354,7 +360,7 @@ public int getResultsPerPage()
> >       */
> >      public void setResultsPerPage(int resultsPerPage)
> >      {
> > -        if (resultsPerPage > -1 || (browseIndex != null && browseIndex.isTagCloudEnabled()))
>
> This change is essentially a revert of the previous commit. Is it
> intentional or were you accidentally working with older source?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DSpace/DSpace/pull/872/files#r25235500>.
>



-- 
Rodrigo Prado de Jesus
http://about.me/royopa
royopa@gmail.com

**Comment by royopa** [comment]
No @helix, this change affect only browse in XMLUI, I did tests earlier.

On 24 February 2015 at 05:37, helix84 <notifications@github.com> wrote:

> Just to be clear - this change affects the default number of items per
> page in browse pages, both XMLUI and JSPUI, both Discovery and Lucene,
> correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DSpace/DSpace/pull/872#issuecomment-75717829>.
>



-- 
Rodrigo Prado de Jesus
http://about.me/royopa
royopa@gmail.com

**Comment by hardyoyo** [comment]
As I commented on the ticket, there is already a configuration for resultsperpage (defaultRpp), it just isn't implemented correctly. I think if we can revamp this PR to utilize the existing config, and not add a new config, I'd approve this PR.

**Comment by royopa** [comment]
Please confirm me.
Does the name of parameter in dspace.cfg is "defaultRpp"? Eg.
# Example results per page
defaultRpp = 10

On Wed, May 20, 2015 at 12:26 PM, Hardy Pottinger <notifications@github.com>
wrote:

> As I commented on the ticket, there is already a configuration for
> resultsperpage (defaultRpp), it just isn't implemented correctly. I think
> if we can revamp this PR to utilize the existing config, and not add a new
> config, I'd approve this PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DSpace/DSpace/pull/872#issuecomment-103925280>.
>



-- 
Rodrigo Prado de Jesus
http://about.me/royopa
royopa@gmail.com

**Comment by hardyoyo** [comment]
https://github.com/DSpace/DSpace/blob/b34240e8d5f315d1a3cad77329c09ffdb34bc8b1/dspace/config/spring/api/discovery.xml#L146

**Comment by royopa** [comment]
Please help me
I must change ConfigurationManager.getIntProperty("webui.browse.results_per_page") to 
Integer.parseInt(ObjectModelHelper.getRequest(objectModel).getParameter("defaultRpp")) ?

**Pull request closed without merging**